lib/SILGen/SILGenLValue.cpp

```diff
@@ -873,9 +873,14 @@ namespace {
           auto &TL = SGF.getTypeLowering(base.getType());
           error = SGF.emitLoad(loc, base.getValue(), TL,
                                SGFContext(), IsNotTake);
+          // Error comes back to us with a +1 cleanup that is not a formal
+          // access cleanup. We need it to be that so that the load nests
+          // properly with other lvalue scoped things like the borrow below.
+          error = SGF.emitFormalAccessManagedRValueWithCleanup(
```

Comment:
As I was looking at this code, I noticed that it is in SILGenLValue where everything /should/ have a formal access duration. I noticed that emitLoad here was always returning a normal load. So, I put in some code here to ensure that we forward that non-formal access release and create a formal access release. That is why in some of the test updates, a destroy_value is hoisted earlier.